### PR TITLE
improve phpdoc

### PR DESCRIPTION
**Pull request by ColonelMoutarde**
Changes proposed in this pull request:

- improve phpdoc

Pull request checklist:

- [x] clear commit messages
- [x] code manually tested
- [x] documentation updated

### CODE REVIEW

File: app/Controllers/configureController.php [hunk 1]
```diff
@@ -16,32 +16,33 @@ public function firstAction(): void {
...
+     *   - display of open action in header
+     *   - display of read action in footer
+     *   - display of favorite action in footer
+     *   - display of sharing action in footer
+     *   - display of article tags in footer
+     *   - display of my Labels in footer
+     *   - display of date in footer
+     *   - display of open action in footer
+     *   - html5 notification timeout (default: 0)
+     * Default values are false unless specified.
```
Comment (Alkarex) [inline app/Controllers/configureController.php]: Please make sure only relevant lines are changed before you commit, as much as possible

Comment (ColonelMoutarde) [reply app/Controllers/configureController.php]: You're right, I did run the linter locally on my commit, but it doesn't seem to be making the changes. I'm running this pr in draft just long enough to fix that.

File: app/Controllers/extensionController.php [hunk 2]
```diff
@@ -52,8 +52,8 @@ protected function getAvailableExtensionList(): array {
 		}
 
 		// fetch the list as an array
-		/** @var array<string,mixed> $list*/
-		$list = json_decode($json, true);
+		/** @var array<string,mixed> $list */
+		$list = json_decode($json);
```
Comment (ColonelMoutarde) [inline app/Controllers/extensionController.php]:
```suggestion
		$list = json_decode($json, true);
```

File: app/Controllers/statsController.php [hunk 3]
```diff
@@ -66,15 +66,15 @@ public function indexAction(): void {
 
 		$feedByCategory = [];
 		$feedByCategory_calculated = $statsDAO->calculateFeedByCategory();
-		for ($i = 0; $i < count($feedByCategory_calculated); $i++) {
+		for ($i = 0, $iMax = count($feedByCategory_calculated); $i < $iMax; $i++) {
```
Comment (Alkarex) [inline app/Controllers/statsController.php]: This does not look good. What do you try to do here?

Comment (Alkarex) [reply app/Controllers/statsController.php]: Ah, is it for an optimisation? Then I suggest either a count do to zero (reverse loop) if the order does not matter (needs to be checked), or a variable before.

Comment (Alkarex) [reply app/Controllers/statsController.php]: It does not look related at all to this PR about improving PHPDoc, though

**Comment by Alkarex** [comment]
Let's be a bit more methodic with https://github.com/FreshRSS/FreshRSS/pull/6037 first

File: app/Controllers/statsController.php (hunk 3 above)
Comment (Alkarex) [reply app/Controllers/statsController.php]: Ping ;-)

File: app/Controllers/feedController.php [hunk 4]
```diff
@@ -344,7 +345,7 @@ public function addAction(): void {
 
 			$feed = $feedDAO->searchByUrl($this->view->feed->url());
 			if ($feed) {
-				// Already subscribe so we redirect to the feed configuration page.
+				// Already subscribe, so we redirect to the feed configuration page.
```
Comment (Alkarex) [inline app/Controllers/feedController.php]:
```suggestion
				// Already subscribed, so we redirect to the feed configuration page.
```

File: app/Controllers/feedController.php [hunk 5]
```diff
@@ -514,13 +516,11 @@ public static function actualizeFeeds(?int $feed_id = null, ?string $feed_url =
 					// Feed is cached and unchanged
 					$newGuids = [];
 					$entries = [];
-					$feedIsEmpty = false;	// We do not know
```
Comment (Alkarex) [inline app/Controllers/feedController.php]: Please avoid mixing unrelated things. This is not related to PHPDoc,
So not for that PR, but even though it was already declared before, it was more explicit this way (or at least a comment should be kept).

**Comment by Alkarex** [comment]
Adding all the `@throws` can probably be done automatically. It would be nice to search for a tool instead of doing too much by hand.
Maybe https://phpstan.org/blog/introducing-phpstan-pro for instance
Input / tests welcome

File: app/Controllers/feedController.php (hunk 4 above)
Comment (Alkarex) [reply app/Controllers/feedController.php]: I still believe there is a typo there

**Comment by Alkarex** [comment]
To supplement my previous messages, I would like a more structured approach than random / partial changes in random files.

For instance, taking one of the files modified with `@throws`, many are missing. So either we do it fully or we do not. At least at file level, or for one type of precise exception.

Setting to `true` the following parameter:
https://github.com/FreshRSS/FreshRSS/blob/7a9a24a0d0f2e5cc674995ce3174116803acde09/phpstan.neon#L42

this gives:

```
$ vendor/bin/phpstan analyse app/Controllers/configureController.php
Note: Using configuration file /home/alexandre/GitHub/FreshRSS/phpstan.neon.
 1/1 [▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓] 100%

 ------ --------------------------------------------------------------------------------------------------------------------------------------------------------------------- 
  Line   configureController.php                                                                                                                                              
 ------ --------------------------------------------------------------------------------------------------------------------------------------------------------------------- 
  :175   Method FreshRSS_configure_Controller::integrationAction() throws checked exception Minz_ConfigurationException but it's missing from the PHPDoc @throws tag.         
  :176   Method FreshRSS_configure_Controller::integrationAction() throws checked exception Minz_ConfigurationException but it's missing from the PHPDoc @throws tag.         
  :251   Method FreshRSS_configure_Controller::archivingAction() throws checked exception FreshRSS_Context_Exception but it's missing from the PHPDoc @throws tag.            
  :252   Method FreshRSS_configure_Controller::archivingAction() throws checked exception FreshRSS_Context_Exception but it's missing from the PHPDoc @throws tag.            
  :260   Method FreshRSS_configure_Controller::archivingAction() throws checked exception FreshRSS_Context_Exception but it's missing from the PHPDoc @throws tag.            
  :261   Method FreshRSS_configure_Controller::archivingAction() throws checked exception FreshRSS_Context_Exception but it's missing from the PHPDoc @throws tag.            
  :262   Method FreshRSS_configure_Controller::archivingAction() throws checked exception FreshRSS_Context_Exception but it's missing from the PHPDoc @throws tag.            
  :273   Method FreshRSS_configure_Controller::archivingAction() throws checked exception FreshRSS_Context_Exception but it's missing from the PHPDoc @throws tag.            
  :274   Method FreshRSS_configure_Controller::archivingAction() throws checked exception FreshRSS_Context_Exception but it's missing from the PHPDoc @throws tag.            
  :283   Method FreshRSS_configure_Controller::archivingAction() throws checked exception FreshRSS_Context_Exception but it's missing from the PHPDoc @throws tag.            
  :285   Method FreshRSS_configure_Controller::archivingAction() throws checked exception Minz_ConfigurationNamespaceException but it's missing from the PHPDoc @throws tag.  
  :285   Method FreshRSS_configure_Controller::archivingAction() throws checked exception Minz_PDOConnectionException but it's missing from the PHPDoc @throws tag.           
  :288   Method FreshRSS_configure_Controller::archivingAction() throws checked exception Minz_ConfigurationNamespaceException but it's missing from the PHPDoc @throws tag.  
  :288   Method FreshRSS_configure_Controller::archivingAction() throws checked exception Minz_PDOConnectionException but it's missing from the PHPDoc @throws tag.           
  :309   Method FreshRSS_configure_Controller::queriesAction() throws checked exception Minz_ConfigurationException but it's missing from the PHPDoc @throws tag.             
  :326   Method FreshRSS_configure_Controller::queriesAction() throws checked exception FreshRSS_Context_Exception but it's missing from the PHPDoc @throws tag.              
  :327   Method FreshRSS_configure_Controller::queriesAction() throws checked exception FreshRSS_Context_Exception but it's missing from the PHPDoc @throws tag.              
  :332   Method FreshRSS_configure_Controller::queriesAction() throws checked exception FreshRSS_Context_Exception but it's missing from the PHPDoc @throws tag.              
  :364   Method FreshRSS_configure_Controller::queryAction() throws checked exception FreshRSS_Context_Exception but it's missing from the PHPDoc @throws tag.                
  :369   Method FreshRSS_configure_Controller::queryAction() throws checked exception FreshRSS_Context_Exception but it's missing from the PHPDoc @throws tag.                
  :400   Method FreshRSS_configure_Controller::queryAction() throws checked exception Minz_ConfigurationException but it's missing from the PHPDoc @throws tag.               
  :415   Method FreshRSS_configure_Controller::queryAction() throws checked exception FreshRSS_Context_Exception but it's missing from the PHPDoc @throws tag.                
  :417   Method FreshRSS_configure_Controller::queryAction() throws checked exception FreshRSS_Context_Exception but it's missing from the PHPDoc @throws tag.                
  :418   Method FreshRSS_configure_Controller::queryAction() throws checked exception FreshRSS_Context_Exception but it's missing from the PHPDoc @throws tag.                
  :431   Method FreshRSS_configure_Controller::deleteQueryAction() throws checked exception FreshRSS_Context_Exception but it's missing from the PHPDoc @throws tag.          
  :436   Method FreshRSS_configure_Controller::deleteQueryAction() throws checked exception FreshRSS_Context_Exception but it's missing from the PHPDoc @throws tag.          
  :438   Method FreshRSS_configure_Controller::deleteQueryAction() throws checked exception FreshRSS_Context_Exception but it's missing from the PHPDoc @throws tag.          
  :439   Method FreshRSS_configure_Controller::deleteQueryAction() throws checked exception FreshRSS_Context_Exception but it's missing from the PHPDoc @throws tag.          
  :453   Method FreshRSS_configure_Controller::bookmarkQueryAction() throws checked exception FreshRSS_Context_Exception but it's missing from the PHPDoc @throws tag.        
  :459   Method FreshRSS_configure_Controller::bookmarkQueryAction() throws checked exception Minz_ConfigurationException but it's missing from the PHPDoc @throws tag.       
  :463   Method FreshRSS_configure_Controller::bookmarkQueryAction() throws checked exception FreshRSS_Context_Exception but it's missing from the PHPDoc @throws tag.        
  :464   Method FreshRSS_configure_Controller::bookmarkQueryAction() throws checked exception FreshRSS_Context_Exception but it's missing from the PHPDoc @throws tag.        
  :493   Method FreshRSS_configure_Controller::systemAction() throws checked exception FreshRSS_Context_Exception but it's missing from the PHPDoc @throws tag.               
  :498   Method FreshRSS_configure_Controller::systemAction() throws checked exception FreshRSS_Context_Exception but it's missing from the PHPDoc @throws tag.               
  :499   Method FreshRSS_configure_Controller::systemAction() throws checked exception FreshRSS_Context_Exception but it's missing from the PHPDoc @throws tag.               
  :500   Method FreshRSS_configure_Controller::systemAction() throws checked exception FreshRSS_Context_Exception but it's missing from the PHPDoc @throws tag.               
  :501   Method FreshRSS_configure_Controller::systemAction() throws checked exception FreshRSS_Context_Exception but it's missing from the PHPDoc @throws tag.               
  :502   Method FreshRSS_configure_Controller::systemAction() throws checked exception FreshRSS_Context_Exception but it's missing from the PHPDoc @throws tag.               
 ------ --------------------------------------------------------------------------------------------------------------------------------------------------------------------- 


                                                                                                                        
 [ERROR] Found 38 errors                                                                                                
```

Furthermore, some of those insights might indicate that we need to think how we handle exceptions in some cases.

P.S.: 889 "errors" of this type in the whole project

File: app/Controllers/statsController.php [hunk 6]
```diff
@@ -67,18 +67,25 @@ public function indexAction(): void {
 
 		$feedByCategory = [];
 		$feedByCategory_calculated = $statsDAO->calculateFeedByCategory();
-		for ($i = 0; $i < count($feedByCategory_calculated); $i++) {
+
+		$iMax = count($feedByCategory_calculated);
+		for ($i = 0; $i < $iMax; $i++) {
```
Comment (Alkarex) [inline app/Controllers/statsController.php:72]: This does not correspond to the topic of the PR on improving PHPDoc (which was already quite vague).
I am not so found of PRs with various random changes, and prefer focussed PRs doing one thing well
